README.md

```diff
@@ -242,6 +242,214 @@ remote_state = {
+of the `.terragrunt` file itself. How do you avoid having to manually maintain the contents of all of these 
+similar-looking `.terragrunt` files?
+
+The solution is to use the following features of Terragrunt:
+
+* Inheritance
+* Find parent helper
+* Relative path helper
+* Overriding settings from the parent
+
+### Inheritance
+
+One `.terragrunt` file can "inherit" from another `.terragrunt` file. For example, imagine you have the following file
+layout:
+
+```
+my-terraform-repo
+  └ .terragrunt
+  └ qa
+      └ my-app
+          └ main.tf
+          └ .terragrunt
+  └ stage
+      └ my-app
+          └ main.tf
+          └ .terragrunt
+  └ prod
+      └ my-app
+          └ main.tf
+          └ .terragrunt
+```
+
+The `.terragrunt` file in the root folder defines the typical `lock` and `remote_state` settings. The `.terragrunt` 
+files in all the subfolders (e.g. `qa/my-app/.terragrunt`) can automatically reuse ("inherit") all the settings from
+a parent file using the `parent` block:
+
+```hcl
+parent = {
+  path = "../../.terragrunt"
+}
+```
+
+There are a few problems with the simple approach above, so read on before using it!
+
+1. Having to manually manage the file paths to the parent folder is tedious. To solve this problem, you can use the 
+   `find_in_parent_folders` helper. 
+1. If the parent `.terragrunt` file hard-codes the `state_file_id` and `key` settings, then every child would rely on
+   the same lock and write state to the same location. To avoid this problem, you can use the `path_relative_to_parent`
+   helper.
+1. Some of the child `.terragrunt` files may want to override settings from the parent. To do this, see the section
+   on overriding settings from the parent.
+
+Each of these items is discussed next.
+
+### find_in_parent_folders helper
+
+Terragrunt supports the use of a few helper functions using the same syntax as Terraform: `${some_function()}`. One of
+the supported helper functions is `find_in_parent_folders`, which returns the path to the first `.terragrunt` file it
+finds in the parent folders above the current `.terragrunt` file. 
+
+Example:
+
+```hcl
+parent = {
+  path = "${find_in_parent_folders()}"
+}
+```
+
+If you ran this in `qa/my-app/.terragrunt`, this would automatically set `path` to `../../.terragrunt`. You will almost
+always want to use this function, as it allows you to copy/paste the same `.terragrunt` file to all child folders with
+no changes.
+
+### path_relative_to_parent helper
+
+Another helper function supported by Terragrunt is `path_relative_to_parent`, which returns the relative path between 
+the current `.terragrunt` file and the path specified in its `parent` block. For example, in the root `.terragrunt` 
+file, you could do the following:
+ 
+```hcl
+# Configure Terragrunt to use DynamoDB for locking
+lock = {
+  backend = "dynamodb"
+  config {
+    state_file_id = "${path_relative_to_parent()}"
```

Comment:
You know, now that I read your confusion, I'm realizing that "inheritance" is probably not a great term. I think a better term is "include". All that I'm doing is I'm including all the content of a parent `.terragrunt` file in the current `.terragrunt` file, just as if that content was copied & pasted. I'll update the docs to try to explain this more clearly.